ai16z/eliza
# chore: improve smokeTests environment validation and logging

# Relates to:
N/A

# Risks
Low - Changes are isolated to the smoke test script and don't affect production code.
- Script improvements are defensive in nature (error handling, timeouts)
- Changes are easily reversible if issues arise

# Background

## What does this PR do?
Enhances the smoke test script with improved logging, error handling, and process management:
- Adds strict mode with error handling (set -euo pipefail)
- Implements process cleanup with trap command (simplifies as well)
- Adds timeout configuration for application startup
- Enhances logging of test output and termination status

## What kind of change is this?
Improvements (misc. changes to existing features)

# Documentation changes needed?
My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?
1. Review the changes in `scripts/smokeTests.sh`
2. Pay particular attention to the error handling and timeout configurations

## Detailed testing steps
1. Run the smoke tests: `pnpm run smokeTests`
2. Verify that version information is displayed at start
3. Verify that the script properly handles timeouts
4. Verify that process cleanup occurs on both success and failure
5. Check that error messages are clear and informative

## Screenshots
N/A - Command line tool changes only

# Deploy Notes
No special deployment steps required - changes only affect development/testing workflow.


@sol.kinetics
